### PR TITLE
API-14243: Only allow requests to developer-portal-backend from developer-portal frontend

### PR DESCRIPTION
**Pull request by SlakrHakr**
### Description

Adding forgery protection to form submissions submitting to new Lighthouse Consumer Management Service in the future.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "homepage": "https://developer.va.gov/",
+  "proxy": "https://blue.prod.lighthouse.va.gov/",
```
Comment (SlakrHakr) [inline package.json:6]: ok, it looks like this configuration only works in development
hoping there is a way in aws to proxy routes over to LCMS?

**Review by gileswells — CHANGES_REQUESTED** [review]

File: src/containers/consumerOnboarding/ProductionAccess.tsx [hunk 2]
```diff
@@ -320,12 +322,19 @@ const ProductionAccess: FC = () => {
         setSubmissionError(true);
       }
       if (flagLpbActive) {
+        const forgeryToken = Math.random().toString(36)
+                                          .substring(2);
```
[the remark this answers is not in the document]
Comment (SlakrHakr) [reply src/containers/consumerOnboarding/ProductionAccess.tsx:326]: The protection comes from both the frontend and backend having access to the same session (being on the same domain).

1. Frontend adds a random string to it's session
2. Frontend passes same random string to the backend in the header of the request
3. Backend validates passed in header against the string in session (that the frontend put there)

[the remark this answers is not in the document]
Comment (SlakrHakr) [reply src/containers/consumerOnboarding/ProductionAccess.tsx:326]: I used rails' implementation of CSRF protection as inspiration for this:
https://medium.com/rubyinside/a-deep-dive-into-csrf-protection-in-rails-19fa0a42c0ef

[the remark this answers is not in the document]
Comment (gileswells) [reply src/containers/consumerOnboarding/ProductionAccess.tsx:326]: I derped on this. 👍

**Pull request merged**